### PR TITLE
feat: add tsconfig.json to basic example

### LINKED ISSUE #3402: vitest example should have explicit `tsconfig.json`
### Clear and concise description of the problem

Currently, there is only one example for pure js/ts testing, and it doesn't have a TS config.

This is an issue for new developers (to typescript or vite) as they don't have much resources to look for basic vite configuration

### Suggested solution

It would be nice if the basic example had explicit `tsconfig.json` and an `src` directory, or at least create another 'advanced' example including the aforementioned concepts and even TS references

### Alternative

_No response_

### Additional context

qol feature

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by allisons11**
### Description

Resolves #3402. Added `tconfig.json` and `src` directory to the basic example to show possible configuration while keeping the example simple.

Included a few basic compiler options in the `tsconfig` (module, target, moduleResolution, strict)  -as well as the `include` and `exclude` fields - to reflect configuration that one might commonly see. Added an example of a function export to `src` to provide an example of testing a module.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: examples/basic/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "compilerOptions": {
+    "module": "esnext",
+    "target": "esnext",
+    "moduleResolution": "node",
```
[the remark this answers is not in the document]
Comment (allisons11) [reply examples/basic/tsconfig.json]: Got it, thanks! 

Updated to follow the strict config, and removed the `moduleResolution` option as it's redundant with `node16` as module.

File: examples/basic/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  "compilerOptions": {
+    "module": "node16",
+    "target": "es2020",
+    "strict": true,
+    "verbatimModuleSyntax": true,
+    "declaration": true,
+    "sourceMap": true,
+    "declarationMap": true
+  },
+  "include": ["src"],
```
Comment (sheremet-va) [inline examples/basic/tsconfig.json:11]: Shouldn't it also cover tests? 🤔

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: examples/basic/tsconfig.json (hunk 2 above)
Comment (allisons11) [reply examples/basic/tsconfig.json]: good catch, definitely should - updated!

**Comment by nopeless** [comment]
thanks!

**Pull request merged**